LoopKit/CarbKit/CarbStore.swift

```diff
@@ -317,8 +323,8 @@ extension CarbStore {
     ///   - samples: An array of samples, in chronological order by startDate
     public func getCachedCarbSamples(start: Date, end: Date? = nil, completion: @escaping (_ samples: [StoredCarbEntry]) -> Void) {
         #if os(iOS)
-        // If we're within our cache duration, skip the HealthKit query
-        guard start <= earliestCacheDate else {
+        // If we're within our observation duration, skip the HealthKit query
+        guard start <= earliestObservationDate else {
             self.queue.async {
                 completion(self.getCachedCarbEntries().filterDateRange(start, end))
```

Comment:
Out of curiosity, what happens if the user has not given permission...is that case handled by the caller?

Reply:
This check is simply asking whether we are in the window of time when we can guarantee that if there is data in HealthKit and we can access it then we definitely have it in Core Data. (The embedded call to `self.getCachedCarbEntries().filterDateRange(start, end)` *only* looks in Core Data.)